api-server/src/server/boot/user.js

```diff
@@ -468,6 +471,93 @@ function postResetProgress(req, res, next) {
   );
 }
 
+function createPostResetProjectProgress(app) {
+  return async function postResetProjectProgress(req, res, next) {
+    const { User } = app.models;
+    const userId = req.user.id;
+    const { blocks } = req.body;
+
+    let toResetChallengeIds = [];
+    let undefinedBlocks = [];
+
+    let isSendBlocks = blocks && blocks.length > 0;
+    let isHasValidBlockToReset = false;
+    let isResetSuccessful = false;
+
+    if (isSendBlocks) {
+      for (const dashedBlockName of blocks) {
+        const blockChallengeIds = getChallengeIdsForBlock(dashedBlockName);
+        if (blockChallengeIds != undefined) {
+          toResetChallengeIds.push(...blockChallengeIds);
+          isHasValidBlockToReset = true;
+        } else {
+          undefinedBlocks.push(dashedBlockName);
+        }
+      }
+
+      if (isHasValidBlockToReset) {
+        try {
+          const result = await User.updateAll(
+            {
+              id: userId,
+              'completedChallenges.id': { in: toResetChallengeIds }
+            },
+            {
+              $pull: {
+                completedChallenges: { id: { in: toResetChallengeIds } }
+              }
+            }
+          );
+          isResetSuccessful = result.count > 0;
+        } catch (err) {
+          return next(err);
+        }
+      }
+    }
+
+    const resetMessage = buildResetMessage(
+      isSendBlocks,
+      isHasValidBlockToReset,
+      isResetSuccessful,
+      undefinedBlocks
+    );
+    return res.status(200).json({ message: resetMessage });
```

Comment:
The status code should probably be a `204` if there were no issues deleting any of the block data.

Otherwise, the status code should be related to the `message`

Reply:
Thanks for the feedback, @ShaunSHamilton! I've adjusted the status codes to reflect the outcome of the operations. For successful resets without any issues, I’ve set the status code to 204 as suggested. The other status codes have been aligned with the corresponding conditions in the response message.